src/util/customErrors.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * A custom error that holds additional information and a more explicit message
+ * to simplify tracking down errors that occur during execution
+ */
+export class AnnotatedError extends Error {
+  // TODO: need to explicitly specify cause in TypeScript versions < 4.7
+  // Remove this when TypeScript is upgraded, and add the cause directly to the `super` call
+  cause: Error;
+
+  constructor(
+    originalError: Error,
+    message: string,
+    public expressionName: string,
+    public libraryName: string,
+    public localId?: string,
+    public locator?: string
```

Comment:
Now that these are properties, they're getting printed out when the error bubbles up:
<img width="870" alt="image" src="https://user-images.githubusercontent.com/2278253/225973952-8f038bf0-4411-4a9c-af76-41781f253f02.png">

This is fairly redundant and made me wonder if we should stop putting them in the message. 

One one hand, if we keep them in the message, then if they're caught in another app using this lib, and that app logs only the message, you still get all the data.

On the other hand, if we let the message just be the message (without all the supplemental data), then we allow consuming apps to make their own decision about how to display the error and associated data to the user (since they can get at it as discrete values). In a web app, this would allow for nicer formatting than our pre-formatted text message. (Although I suppose they could still do that by _ignoring_ `e.message` and using `e.cause.message` instead).

What do you think?

Reply:
What if we removed it from the message, but maybe some type of primer about how additional info may be available for annotated errors.

Should we document this somehow as well? Maybe in a debugging section in the README or something? I can see how these properties could be hidden by someone who doesn't know they're supposed to be there, so I think some type of indication that _some_ additional info is there in the message would be nice.

Reply:
I honestly don't know what the right answer is here. But yeah, if we choose to go w/ the simple message and leave display of the details to the consumer, then I think it makes a lot of sense to document it. Good idea.

Do you think we should do it?

Reply:
OK. We just agreed over slack that since we're both conflicted on this, it's best to go with the path of least resistance -- which is to leave it as it is now.